libgalois/include/katana/gstl.h

```diff
@@ -40,51 +40,134 @@ namespace katana {
+//! main thread always does the deallocation (after the parallel region).
 template <typename T>
-using Deque = std::deque<T, Pow2Alloc<T>>;
-
-template <typename T>
-using List = std::list<T, FixedSizeAlloc<T>>;
-
+using List = std::list<T, FixedSizeAllocator<T>>;
+
+//! [STL set using katana Pow2BlockAllocator]
+//! specializes std::set to use katana concurrent, scaleable allocator:
+//! the allocator is composed of thread-local allocators
+//! that manage thread-local pages and only use a global lock to
+//! get allocation in chunks of (huge) pages.
+//! When destroyed, its memory is reclaimed by the thread that destroys it.
+//!
+//! Use this when allocations and deallocations can occur in a parallel region.
+//! As the memory allocated can be reused for another allocation only by the
+//! thread that deallocated it, this is not suitable for use cases where the
+//! main thread always does the deallocation (after the parallel region).
 template <typename T, typename C = std::less<T>>
-using Set = std::set<T, C, Pow2Alloc<T>>;
-
+using Set = std::set<T, C, Pow2BlockAllocator<T>>;
+
+//! [STL unordered_set using katana Pow2BlockAllocator]
+//! specializes std::unordered_set to use katana concurrent, scaleable allocator:
+//! the allocator is composed of thread-local allocators
+//! that manage thread-local pages and only use a global lock to
+//! get allocation in chunks of (huge) pages.
+//! When destroyed, its memory is reclaimed by the thread that destroys it.
+//!
+//! Use this when allocations and deallocations can occur in a parallel region.
+//! As the memory allocated can be reused for another allocation only by the
+//! thread that deallocated it, this is not suitable for use cases where the
+//! main thread always does the deallocation (after the parallel region).
 template <
     typename T, typename Hash = std::hash<T>,
     typename KeyEqual = std::equal_to<T>>
-using UnorderedSet = std::unordered_set<T, Hash, KeyEqual, Pow2Alloc<T>>;
-
+using UnorderedSet =
+    std::unordered_set<T, Hash, KeyEqual, Pow2BlockAllocator<T>>;
+
+//! [STL map using katana Pow2BlockAllocator]
+//! specializes std::map to use katana concurrent, scaleable allocator:
+//! the allocator is composed of thread-local allocators
+//! that manage thread-local pages and only use a global lock to
+//! get allocation in chunks of (huge) pages.
+//! When destroyed, its memory is reclaimed by the thread that destroys it.
+//!
+//! Use this when allocations and deallocations can occur in a parallel region.
+//! As the memory allocated can be reused for another allocation only by the
+//! thread that deallocated it, this is not suitable for use cases where the
+//! main thread always does the deallocation (after the parallel region).
 template <typename K, typename V, typename C = std::less<K>>
-using Map = std::map<K, V, C, Pow2Alloc<std::pair<const K, V>>>;
-
+using Map = std::map<K, V, C, Pow2BlockAllocator<std::pair<const K, V>>>;
```

Comment:
Why are we not using FixedSizeAllocator for std::map and std::set?

Reply:
No idea. I can change it if you think that's more appropriate.

Reply:
Please do. My understanding is that since it's a Red-black tree, the allocations are always fixed size